### PR TITLE
Document Deployment Scenarios #119 [skip ci]

### LINKED ISSUE #119: Suggestion: Document improvement
Hi,
Please can you remove references to ‘cluster’ from JanusGraph documentation, specifically on the Graph Partitioning page – this is one thing that threw me with Titan, as there is no 'cluster'. There are several isolated standalone services.

I would also love to see a recommended/suggested/example production architecture diagram (not just words please), with high availability etc, for both the general case and possibly specific to each backend store.

Many thanks!
John

### PR DESCRIPTION
**Pull request by FlorianHockmann**
The added chapter Deployment Scenarios showcases different deployment scenarios for JanusGraph with their advantages and disadvantages to help especially new users find a good deployment option for their environment.

The two changes in the chapter Graph Partitioning are directly taken from @robertdale's suggestions in #119 to reduce the confusion of what is meant by _cluster_.

It probably makes sense to also change the chapters for Cassandra and HBase as they contain similar diagrams to explain their configuration but I wanted to make sure first that the diagrams I added make sense like that before I change any other chapters to also use them / link to them.

I created the diagrams with [draw.io](https://www.draw.io/) and included the respective XML files so they can be imported and edited.

A **preview** of the new chapter is available [here](https://florianhockmann.github.io/docs.janusgraph.org/0.2.2-SNAPSHOT/deployment-scenarios.html).

Fixes #119

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pluradj** [comment]
nice, thanks @FlorianHockmann! can you please retarget this for `0.2` branch. thanks.

File: docs/deploymentscenarios.adoc [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+It is also possible to host JanusGraph Server together with the backend(s) on just one server. This is especially attractive for testing purposes or for example when JanusGraph just supports a single application which can then also run on the same server.
+
+image:minimalist-scenario.svg[Minimalist deployment scenario diagram, 650]
+
+Opposed to the previous scenarios, it makes most sense to use backends for this scenario that are not scalable. The in-memory backend can be used for testing purposes or Berkeley DB for production and Lucene as the optional index backend.
+
+[[embedded-janusgraph]]
+=== Embedded JanusGraph
+
+Instead of connecting to the JanusGraph Server from an application it is also possible to embed JanusGraph as a library inside a JVM based application. While this reduces the administrative overhead it makes it impossible to scale JanusGraph independently of the application.
```
Comment (pluradj) [inline docs/deploymentscenarios.adoc]: missing a comma in here:

While this reduces the administrative overhead, it makes it impossible to scale JanusGraph independently of the application.

**Comment by FlorianHockmann** [comment]
Thanks for your comments so far, @pluradj. I just retargeted the PR for the `0.2` branch, added the missing comma and published a [preview of the docs](https://florianhockmann.github.io/docs.janusgraph.org/0.2.2-SNAPSHOT/deployment-scenarios.html).

**Comment by pluradj** [comment]
I think the ordering is fine as written. The Getting Started scenario is essentially the same as the pre-packaged distribution with the Gremlin Server, single node Cassandra, and single node Elasticsearch described in [Chapter 7](https://docs.janusgraph.org/latest/server.html).

The minimalist scenario is probably the least adopted because, based on mailing list and GitHub issues traffic, Berkeley and Lucene don't get many mentions.

File: docs/deploymentscenarios.adoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
+[[deployment-scenarios]]
+== Deployment Scenarios 
+
+JanusGraph offers a wide choice of storage and index backends which results in great flexibility of how it can be deployed. This chapter presents a few possible deployment scenarios to help with the complexity that comes with this flexibility.
+
+Before discussing the different deployment scenarios, it is important to understand the roles of JanusGraph itself and that of the backends. First of all, applications only communicate directly with JanusGraph, mostly by sending Gremlin traversals for execution. JanusGraph then communicates with the configured backends to execute the received traversal. Since there is nothing like a _master_ JanusGraph Server, the applications can connect to any JanusGraph Server instance. They can also use a load-balancer to schedule requests to the different instances. The JanusGraph Server instances themselves don't communicate to each other directly which makes it easy to scale JanusGraph when the need arises to process more traversals.
```
Comment (jerryjch) [inline docs/deploymentscenarios.adoc]: > the roles of JanusGraph itself and that of the backends. First of all, applications only communicate directly with JanusGraph, mostly by sending Gremlin traversals for execution. JanusGraph then communicates with the configured backends to execute the received traversal.

We seem to be using the term 'JanusGraph' and 'JanusGraph server' intermingly.  Can we clarify?

**Comment by FlorianHockmann** [comment]
@jerryjch 
>Let's add a statement somewhere at the first paragraph to say that these are a few deployment examples and each user's deployment choice lies in the use cases and production need.

I just added a `NOTE` which should explain that.

> We seem to be using the term 'JanusGraph' and 'JanusGraph server' intermingly. Can we clarify?

You're right that's a bit confusing. The explanations in this paragraph mostly apply to JanusGraph in general, irrespective of whether it is used as JanusGraph Server or embedded. I tried to make it a bit more clear why the text suddenly talks about JanusGraph Server:

> When JanusGraph is used in the form of JanusGraph Server, then there is nothing like a master JanusGraph Server.

Are you ok with the changes like this or do have other suggestions on how to improve these aspects?

**Comment by pluradj** [comment]
LGTM, thanks for the revisions @FlorianHockmann.

**Pull request merged**